pkg/csi/recover/recover.go

```diff
@@ -151,20 +154,36 @@ func (r *FuseRecover) runOnce() {
 	r.recover()
 }
 
-func (r FuseRecover) recover() {
+func (r *FuseRecover) recover() {
 	brokenMounts, err := mountinfo.GetBrokenMountPoints()
 	if err != nil {
 		glog.Error(err)
 		return
 	}
 
 	for _, point := range brokenMounts {
-		glog.V(4).Infof("Get broken mount point: %v", point)
+		if lock := r.locks.TryAcquire(point.MountPath); !lock {
+			glog.V(4).Infof("FuseRecovery: fail to acquire lock on path %s, skip recovering it", point.MountPath)
+			continue
+		}
+
+		should, err := r.shouldRecover(point.MountPath)
+		if err != nil {
+			glog.Warningf("FuseRecovery: found path %s which is unable to recover due to error %v, skip it", point.MountPath, err)
+			continue
```

Comment:
If `continue` here, the lock of mountPath will not release.

Reply:
Fixed. Thx for pointing this!